pkg/substitution/substitution.go

```diff
@@ -199,33 +214,30 @@ func ValidateVariableIsolatedP(value, prefix string, vars sets.String) *apis.Fie
 	return nil
 }
 
-// ValidateWholeArrayOrObjectRefInStringVariable validates if a single string field uses references to the whole array/object appropriately
+// ValidateIsolatedArrayOrObjectRefInStringVariable validates if a single string field uses references to the whole array/object appropriately
```

Comment:
what is an isolated reference?

Reply:
An isolated reference means a string that contains only a reference to a variable but nothing else i.e. no extra characters, no reference to other variables. Some examples

- Isolated reference: `"$(params.myParam)"`, `"$(params['myArray'][*])"`
- Non-isolated reference: `"this is param $(params.myParam)"`, `"the first param is $(params.first) and the second param is $(params["second"])"`

Reply:
Makes sense! could you add that as a comment in the code?

Reply:
SG! Added! Thanks Lee.